src/ws/Socket.ts

```diff
@@ -684,4 +686,14 @@ export class Socket extends EventEmitter {
             this.status = Socket.CLOSED;
         }
     }
+
+    private callAuth(args: authArgs, options?: ICallOptions) {
+        return this.call(authMethod, args, options).catch((err: IErrorPacket) => {
```

Comment:
nit: can this be done with a try/catch and await?

Reply:
async/await expands horribly when targeting es5, unfortunately